eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/ZmqWatcher.scala

```diff
@@ -251,6 +247,50 @@ class ZmqWatcher(chainHash: ByteVector32, blockCount: AtomicLong, client: Extend
     }
   }
 
+  // TODO: move to a separate actor that listens to CurrentBlockCount and manages utxos for RBF
+  def checkUtxos(): Future[Unit] = {
```

Comment:
Making this method returns a `Future` is good because it will save the parent watcher actor from failing in case of problems.

However I think an `onFailure` or `recover` with a `log.warn` would be nice to detect if such problems happen.

Reply:
Definitely, good idea.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1609/commits/43d726be7184e941f80b10b94f541fc8f239e4e3

Reply:
This the build is failing, I'd rather do the `recover` in the method itself, it's more consistent with the rest of the file.

Reply:
Fixed in https://github.com/ACINQ/eclair/pull/1609/commits/59b9e32e6eec41d6afae9426a6d514a710ba458b :sweat_smile: